flask/src/main/java/net/corda/gradle/flask/FlaskJarTask.java

```diff
@@ -0,0 +1,239 @@
+package net.corda.gradle.flask;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.SneakyThrows;
+import net.corda.flask.common.Flask;
+import net.corda.flask.common.ManifestEscape;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.internal.file.CopyActionProcessingStreamAction;
+import org.gradle.api.internal.file.copy.CopyAction;
+import org.gradle.api.internal.file.copy.CopyActionProcessingStream;
+import org.gradle.api.internal.file.copy.FileCopyDetailsInternal;
```

Comment:
I can't figure out how do I compute the manifest first if I fallback to `org.gradle.api.tasks.bundling.Jar`.. I can't seem to find any way to get the list of files that are to be included in the archive before they are actually written (I need them to generate the manifest file). I could even remove everything from the manifest file and just create my own file under META-INF, but still I need to add some kind of callback after the task configuration phase but before the execution phase that is provided with the list of files that will be included in the generated archive.

(btw I copied the current approach from the shadow plugin that [has been using it for 7 years](https://github.com/johnrengelman/shadow/blame/b8b516d60deeac82df1800fe9abbedb04ff5313a/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowCopyAction.groovy#L28), I know it could break in future versions of Gradle but it doesn't seem too vulnerable to me)